### ISSUE TITLE
Compatibility with Mega 2560 board

### ISSUE THREAD

**Issue by smellykid101** [Flydroid/ADS12xx-Library#4]
I'm having a difficult time communicating to ADS1248 with my Arduino Mega 2560 board. I've set the following pins:
START adc pin 23 -----> arduino pin  47
CS adc pin 24 -----------> arduino pin 53
DRDY adc pin 26 -------> arduino pin 50
SCLK adc pin 28 -------> arduino pin 52
DIN adc pin 27 ----------> arduino pin 51
DVDD adc pin 1 ------->  5V
DGND adc pin 2 -------> gnd
AVDD adc pin 22 -------> 5V
AVSS adc pin 21 -------> gnd

Every time I run the example sketch I see the Arduino responding but I don't have any response from the adc in the serial monitor. I've been stuck and out of ideas what and how to troubleshoot. Any suggestions?

Simon

**Comment by Flydroid** [comment]
You named the DOUT pin the DRDY pin. 
The DOUT and DRDY are different pins. But there is to possibility to have DRDY functionality on the DOUT pin, but this isn't configured in my example. DOUT is the output of the SPI bus. The DRDY pin signals if a conversion is ready. If you look at my code in ad12xx.cpp you can see a WaitforDRDY() function. As some commands have to be issued when DRDY is low. When you give a command and nothing happens, always check first if your DRDY connection is ok.
so connect: DOUT adc pin 26 ---> arduino pin 50
and DRDY adc pin 26  ---> to any digital arduino input

**Comment by smellykid101** [comment]
You're right, I mistakenly wrote DRDY where I meant DOUT. After your suggestion to take a closer look at the WaitforDRDY() function, I realized this is where the code sits because DRDY never goes LOW. I've confirmed with a multimeter that as soon as I give power to the digital pins, DRDY sits at 5V, no matter what I try. Any suggestions? Perhaps a way to confirm its a defective chip?

**Comment by Flydroid** [comment]
A multimeter is probably not fast enough to detected the change on DRDY
signal. Try to get access to an oscilloscope. Without it its really hard to
know if the signal is changing.
Also if the scope can measure the frequency of the changes, you can confirm
that your register write works it rate of the DRDY signal corresponds to
you selected data rate in the SYS0 register

I had also connections problems to the chip while i was using a
breakoutboard in my breadboard setup. The problem were dirty/bad solder
joints on the breakoutboard.
Best use some solder flux and clean it after with alcohol.

I spend a long time getting the communication to work. A small logic
analyzer like that helped a lot
https://www.ikalogic.com/scanaquad-logic-analyzer-signal-generators/

2016-02-26 8:30 GMT+01:00 smellykid101 notifications@github.com:

**Comment by smellykid101** [comment]
I found I needed to set a few pins high/low (I think reset and clock) which let me obtain conversions from the adc. I haven't confirmed all the registers write correctly but SYS0 does. I connected an led to DRDY and it blinks at various sample rates I set. I might get my hands on an oscilloscope next week so I'll try to confirm everything.

The logic analyzer should would be helpful, might pick one up. Sorry for the late reply, super busy with thesis work. Thanks again for the quick advice, this library helped a lot!
